tripy/nvtripy/frontend/cache.py

```diff
@@ -0,0 +1,162 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import hashlib
+from typing import List, Dict, Optional
+
+import mlir_tensorrt.runtime.api as runtime
+
+from nvtripy import utils
+from nvtripy import config
+
+
+class ExecutableCache:
+    """Global cache for storing compiled executables."""
+
+    def __init__(self):
+        self._cache: Dict[str, runtime.Executable] = {}
+
+    def _assign_tensor_name(
+        self,
+        tensor: "tripy.frontend.trace.tensor.TraceTensor",
+        tensor_map: Dict[int, str],
+        next_id: List[int],
+        backup_map: Dict[int, str] = None,
+    ) -> str:
+        """
+        Assign or retrieve a tensor name.
+
+        Args:
+            tensor (TraceTensor): The tensor to name.
+            tensor_map (Dict[int, str]): Mapping of tensor ids to names (clean or original).
+            next_id (List[int]): Mutable list for tracking next tensor ID.
+            backup_map (Dict[int, str], optional): Mapping to store original names. Defaults to None.
+
+        Returns:
+            str: The assigned or retrieved tensor name.
+        """
+        t_id = id(tensor)
+
+        # If tensor not in map, assign new name
+        if t_id not in tensor_map:
+            new_name = f"t{next_id[0]}"
```

Comment:
Is `next_id` actually necessary? Can we just use `len(tensor_map)`? That would simplify a lot of the call-sites.

Reply:
sg! removed the id